vscode-dotnet-sdk-extension/src/extension.ts

```diff
@@ -0,0 +1,150 @@
+    ErrorConfiguration,
+    ExtensionConfigurationWorker,
+    formatIssueUrl,
+    IDotnetAcquireContext,
+    IDotnetUninstallContext,
+    IEventStreamContext,
+    IExtensionContext,
+    IIssueContext,
+    InstallationValidator,
+    registerEventStream,
+    VersionResolver,
+    WindowDisplayWorker,
+} from 'vscode-dotnet-runtime-library';
+import { dotnetCoreAcquisitionExtensionId } from './DotnetCoreAcquistionId';
+
+// Extension constants
+namespace configKeys {
+    export const installTimeoutValue = 'installTimeoutValue';
+    export const enableTelemetry = 'enableTelemetry';
+}
+namespace commandKeys {
+    export const acquire = 'acquire';
+    export const uninstallAll = 'uninstallAll';
+    export const showAcquisitionLog = 'showAcquisitionLog';
+    export const reportIssue = 'reportIssue';
+}
+const commandPrefix = 'dotnet-sdk';
+const configPrefix = 'dotnetSDKAcquisitionExtension';
+const displayChannelName = '.NET SDK';
+const defaultTimeoutValue = 240;
+
+export function activate(context: vscode.ExtensionContext, extensionContext?: IExtensionContext) {
+    const extensionConfiguration = extensionContext !== undefined && extensionContext.extensionConfiguration ?
+        extensionContext.extensionConfiguration :
+        vscode.workspace.getConfiguration(configPrefix);
+    const extension = vscode.extensions.getExtension(dotnetCoreAcquisitionExtensionId);
+
+    if (!extension) {
+        throw new Error(`Could not resolve dotnet acquisition extension '${dotnetCoreAcquisitionExtensionId}' location`);
+    }
+
+    const eventStreamContext = {
+        displayChannelName,
+        logPath: context.logPath,
+        extensionId: dotnetCoreAcquisitionExtensionId,
+        enableTelemetry: enableExtensionTelemetry(extensionConfiguration, configKeys.enableTelemetry),
+        telemetryReporter: extensionContext ? extensionContext.telemetryReporter : undefined,
+    } as IEventStreamContext;
+    const [eventStream, outputChannel, loggingObserver, eventStreamObservers] = registerEventStream(eventStreamContext);
+
+    const displayWorker = extensionContext ? extensionContext.displayWorker : new WindowDisplayWorker();
+    const extensionConfigWorker = new ExtensionConfigurationWorker(extensionConfiguration, undefined);
+    const issueContext = (errorConfiguration: ErrorConfiguration | undefined, commandName: string, version?: string) => {
+        return {
+            logger: loggingObserver,
+            errorConfiguration: errorConfiguration || AcquireErrorConfiguration.DisplayAllErrorPopups,
+            displayWorker,
+            extensionConfigWorker,
+            eventStream,
+            commandName,
+            version,
+        } as IIssueContext;
+    };
+    const timeoutValue = extensionConfiguration.get<number>(configKeys.installTimeoutValue);
+    let storagePath: string;
+    if (os.platform() === 'win32') {
+        storagePath = process.env.APPDATA ? process.env.APPDATA : context.globalStoragePath;
```

Comment:
Should leave a comment here as to why we utilize appdata as our storage path on windows